test/unit/app/controllers/transactions/pending-tx-tracker-test.js

```diff
@@ -0,0 +1,616 @@
+      const warningListener = sinon.spy()
+
+      pendingTxTracker.on('tx:warning', warningListener)
+      pendingTxTracker.resubmitPendingTxs('0x1')
+
+      assert.ok(getPendingTransactions.calledOnceWithExactly(), 'should call getPendingTransaction')
+      assert.ok(resubmitTx.calledTwice, 'should call _resubmitTx')
+      assert.ok(warningListener.notCalled, "should NOT emit 'tx:warning'")
+    })
+
+    it("should NOT emit 'tx:warning' for known failed resubmission", async function () {
+      const getPendingTransactions = sinon.stub().returns([{
+        id: 1,
+      }])
+      const pendingTxTracker = new PendingTransactionTracker({
+        query: {
+          getTransactionReceipt: sinon.stub(),
+        },
+        nonceTracker: {
+          getGlobalLock: sinon.stub().resolves({
+            releaseLock: sinon.spy(),
+          }),
+        },
+        getPendingTransactions,
+        getCompletedTransactions: sinon.stub().returns([]),
+        approveTransaction: sinon.spy(),
+        publishTransaction: sinon.spy(),
+        confirmTransaction: sinon.spy(),
+      })
+      const resubmitTx = sinon.stub(pendingTxTracker, '_resubmitTx').rejects({ message: 'known transaction' })
+      const warningListener = sinon.spy()
+
+      pendingTxTracker.on('tx:warning', warningListener)
+      pendingTxTracker.resubmitPendingTxs('0x1')
+
+      assert.ok(getPendingTransactions.calledOnceWithExactly(), 'should call getPendingTransaction')
+      assert.ok(resubmitTx.calledOnce, 'should call _resubmitTx')
+      assert.ok(warningListener.notCalled, "should NOT emit 'tx:warning'")
+    })
+  })
+
+  describe('#updatePendingTxs', function () {
+    it('should call _checkPendingTx for each pending transaction', async function () {
+      const txMeta = {
+        id: 1,
+        hash: '0x0593ee121b92e10d63150ad08b4b8f9c7857d1bd160195ee648fb9a0f8d00eeb',
+        status: 'signed',
+        txParams: {
+          from: '0x1678a085c290ebd122dc42cba69373b5953b831d',
+          nonce: '0x1',
+          value: '0xfffff',
+        },
+        history: [{}],
+        rawTx: '0xf86c808504a817c800827b0d940c62bb85faa3311a998d3aba8098c1235c564966880de0b6b3a7640000802aa08ff665feb887a25d4099e40e11f0fef93ee9608f404bd3f853dd9e84ed3317a6a02ec9d3d1d6e176d4d2593dd760e74ccac753e6a0ea0d00cc9789d0d7ff1f471d',
+      }
+      const txList = [1, 2, 3].map((id) => ({ ...txMeta, id }))
+      const pendingTxTracker = new PendingTransactionTracker({
+        query: {
+          getTransactionReceipt: sinon.stub(),
+        },
+        nonceTracker: {
+          getGlobalLock: async () => {
+            return { releaseLock: () => {} }
+          },
+        },
+        getPendingTransactions: () => txList,
+        getCompletedTransactions: () => {
+          return []
+        },
+        publishTransaction: () => {},
+        confirmTransaction: () => {},
+      })
+
+      const checkPendingTxStub = sinon.stub(pendingTxTracker, '_checkPendingTx').resolves()
```

Comment:
I'm not sure how useful this style of test is, since all it's doing is confirming an internal method is being called. It doesn't actually test what the function _does_, and it presents an obstacle to refactoring.

Reply:
Oh, this private function is tested separately. As are many other private functions.

Not a huge fan of testing private functions either, for the same reasons. Oh well. I guess this isn't new as of this PR, since they were already tested (just not as well).